localstack/services/apigateway/apigateway_listener.py

```diff
@@ -118,9 +118,12 @@ def forward_request(self, method, path, data, headers):
                     response = Response()
                     parsed_result = json.loads(result)
                     response.status_code = int(parsed_result['statusCode'])
-                    response.headers.update(parsed_result['headers'])
-                    response_body = parsed_result['body']
-                    response._content = json.dumps(response_body)
+                    response.headers.update(parsed_result.get('headers', {}))
+                    try:
+                        response_body = parsed_result['body']
```

Comment:
Ok makes sense, thanks for the update. In that case we should at least make sure that a default value is assigned to `response._content` (just an empty string should be fine), because if that variable is `None`, it may potentially lead to problems downstream (in `generic_proxy.py`